src/org/terraform/coregen/HeightMap.java

```diff
@@ -99,6 +100,16 @@ public static double getMountainousHeight(TerraformWorld tw, int x, int z) {
     }
 
     public static double getPreciseHeight(TerraformWorld tw, int x, int z) {
+        Pair<Integer, Integer> pair = new Pair<>(ChunkCache.getChunkCoordinate(x), ChunkCache.getChunkCoordinate(z));

```

Comment:
Instead of importing the whole JavaFX library... Use MegaChunk class.
Its `hashCode()` and `equals()` are optimized specifically for chunks.